test/test_cpp_standards.py

```diff
@@ -35,28 +114,48 @@ def test_cpp14():
     assert set(actual_wheels) == set(expected_wheels)
 
 
-def test_cpp17():
+cpp17_project = cpp_template_project.copy()
+
+if utils.platform == 'windows':
+    cpp17_project.template_context['extra_compile_args'] = ['/std:c++17', '/wd5033']
+else:
+    cpp17_project.template_context['extra_compile_args'] = ['-std=c++17', '-Wno-register']
+
+cpp17_project.template_context['spam_cpp_top_level_add'] = r'''
+#include <utility>
+auto a = std::pair(5.0, false);
+'''
+
+
+def test_cpp17(tmp_path):
     # This test checks that the C++17 standard is supported
+    project_dir = tmp_path / 'project'
 
-    # Python and PyPy 2.7 use the `register` keyword which is forbidden in the C++17 standard
-    # The manylinux1 docker image does not have a compiler which supports C++11
+    cpp17_project.generate(project_dir)
+
+    # - Python and PyPy 2.7 use the `register` keyword which is forbidden in
```

Comment:
I guess we should move this closer to where the `/wd5033`/`-Wno-register` flags are added.

Reply:
But PyPy and Python 2.7 are skipped - I thought that's what the comment was explaining?

Reply:
No, the main problem is the default old compiler that is used by 2.7. The `register` keyword removed in C++17 but occuring in Python 2's headers is/can be ignored/suppressed by passing `/wd5033`/`-Wno-register`.

But since we're not building 2.7, they can actually be removed, I think!

Reply:
My mistake - python 2.7 is still built on mac/linux. So those flags are still needed

Reply:
Ah, of course! Sorry, forgot about that as well when I thought it could be removed!